symbolic/src/Data/Macaw/Symbolic/Stack.hs

```diff
@@ -103,11 +110,127 @@ createArrayStack bak mem slots sz = do
   (base, mem1) <- mallocStack bak mem sz
   mem2 <- CLM.doArrayStore bak mem1 base stackAlign arr sz
 
-  end <- CLM.doPtrAddOffset bak mem2 base sz
+  off <- WI.bvSub sym sz =<< WI.bvOne sym ?ptrWidth
+  end <- CLM.doPtrAddOffset bak mem2 base off
```

Comment:
Can you leave a comment explaining what is going on here? This ended up being surprisingly subtle, and I'll admit that I'm not entirely sure why this part is necessary, despite knowing that it was important for fixing the test suite.